File: Android/app/src/main/java/com/aram/chatproto/Model/Message.java
package com.aram.chatproto.Model;

import com.google.firebase.database.DataSnapshot;

/**
 * Created by Aram on 2018. 1. 25..
 */

public class Message {
    String fromId;
    String original_text;
    String text;
    String toId;
    int type;
    String timestamp;
    String id;

    public String getOptions() {
        return options != null ? options : "";
    }

    public String getDescription() {
        return description != null ? description : "";
    }

    String options;
    String description;
    public String getFromId() {
        return fromId;
    }

    public String getOriginal_text() {
        return original_text;
    }

    public String getText() {
        return text;
    }

    public String getToId() {
        return toId;
    }

    public int getType() {
        return type;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public String getId() {
        return id;
    }

    public Message(DataSnapshot dataSnapshot) {
        id = dataSnapshot.getKey();
        fromId = dataSnapshot.child("fromId").getValue().toString();
        toId = dataSnapshot.child("toId").getValue().toString();
        text = dataSnapshot.child("text").getValue().toString();
        type = Integer.valueOf(dataSnapshot.child("type").getValue().toString());
        timestamp = dataSnapshot.child("timestamp").getValue().toString();
        if(type == 2){
            description = dataSnapshot.child("description").getValue().toString();
            options = dataSnapshot.child("options").getValue().toString();
        }
        if(dataSnapshot.hasChild("original_text")){
            original_text = dataSnapshot.child("original_text").getValue().toString();
        }
    }
}

File: iOS/ChatForEnglish/ChatForEnglish/Model/Message.swift
//
//  Message.swift
//  ChatForEnglish
//
//  Created by ARam on 2018. 1. 11..
//  Copyright © 2018년 aram. All rights reserved.
//

class Message {
    var id : String?
    var toId : String?
    var fromId : String?
    var timestamp : Int64?
    var text : String?
    var type : Int64?
    var original_text : String?
    var options : String?
    var description: String?
    
    init(values: [String : AnyObject]){
        
        id = values["id"] as? String
        toId = values["toId"] as? String
        fromId = values["fromId"] as? String
        timestamp = values["timestamp"] as? Int64
        text = values["text"] as? String
        type = values["type"] as? Int64
        
        if type == 2 {
            original_text = values["original_text"] as? String
            options = values["options"] as? String
            description = values["description"] as? String
            
        }
        

    }
    
}
